lib/plugin-api/hydrogen-provider.js

```diff
@@ -11,31 +11,38 @@
 * learning how to interact with Hydrogen in your own plugin.
 *
 * @class HydrogenProvider
-###
-module.exports =
-class HydrogenProvider
-    constructor: (@_hydrogen) ->
-        @_happy = true
+*/
+export default class HydrogenProvider {
+    constructor(_hydrogen) {
+        this._hydrogen = _hydrogen;
+        this._happy = true;
+    }
 
-    ###
+    /*
     * Calls your callback when the kernel has changed.
     * @param {Function} Callback
-    ###
-    onDidChangeKernel: (callback) ->
-        @_hydrogen.emitter.on 'did-change-kernel', (kernel) ->
-            if kernel?
-                callback kernel.getPluginWrapper()
-            else
-                callback null
+    */
+    onDidChangeKernel(callback) {
+        return this._hydrogen.emitter.on('did-change-kernel', function(kernel) {
+            if (kernel != null) {
+                return callback(kernel.getPluginWrapper());
+            } else {
+                return callback(null);
+            }
+        });
+    }
 
-    ###
+    /*
     * Get the `HydrogenKernel` of the currently active text editor.
     * @return {Class} `HydrogenKernel`
-    ###
-    getActiveKernel: ->
-        unless @_hydrogen.kernel?
-            grammar = @_hydrogen.editor.getGrammar()
-            language = @_hydrogen.kernelManager.getLanguageFor grammar
-            throw new Error "No running kernel for language `#{language}` found"
+    */
+    getActiveKernel() {
+        if (this._hydrogen.kernel == null) {
+            let grammar = this._hydrogen.editor.getGrammar();
+            let language = this._hydrogen.kernelManager.getLanguageFor(grammar);
+            throw new Error(`No running kernel for language \`${language}\` found`);
+        }
 
-        return @_hydrogen.kernel.getPluginWrapper()
+        return this._hydrogen.kernel.getPluginWrapper();
```

Comment:
Name clashes are definitely a real issue, but the pattern you suggest is not an ideal solution either. If you listen to `onDidChangeKernel` events, you get wrapper objects and not any custom struct. So each plug in will need to end up rolling their own dictionary lookup to fix that. I think adding some methods to encourage better namespacing (e.g. `getData (namespace)`), or figuring out how to return a different set of wrapper objects to each consumer, are more promising approaches.